cfug/dio
# Improved documentation

This adds doc pages to the existing API docs. This is similar to the approach of the docs of GoRouter, see for example the following page: https://pub.dev/documentation/go_router/latest/topics/Get%20started-topic.html

This works by making use of categories (also called topics). See dartdoc explanation at https://pub.dev/packages/dartdoc#advanced-features

### How to test this

Execute `dart doc` in the dio folder. This creates the docs. When opening the index.html, you should see the newly added migration page in the API docs.

### Further stops

I'll try to migrate further documentation to, but further I'd like some feedback.

### New Pull Request Checklist

- [x] I have read the [Documentation](https://pub.dev/documentation/dio/latest/)
- [x] I have searched for a similar pull request in the [project](https://github.com/cfug/dio/pulls) and found none
- [x] I have updated this branch with the latest `main` branch to avoid conflicts (via merge from master or rebase)
- [ ] I have added the required tests to prove the fix/feature I'm adding
- [x] I have run the tests without failures
- [ ] I have updated the `CHANGELOG.md` in the corresponding package

### Additional context and info (if any)



Fixes #1921
